### ISSUE TITLE
Inheritance in TG.Program

### ISSUE THREAD

**Issue by fernandojsg** [mrdoob/texgen.js#20]
Every TG.Program needs the same two functions `getParams`and `setParamValue`.

``` javascript
getParams: function () {
    return params;
},
setParamValue: function ( id, value ) {
    params[ id ] = value;
},
```

Could be nice to be implemented just in TG.Program, not on every subclass.

**Comment by mrdoob** [comment]
Yeah, the problem is that inheritance works backwards now... `TG.Program` adds stuff to the program. In order to implement the _proper_ way the code becomes a bit more verbose but I'll see if I can find a way :+1:
